report-generator/src/main/java/com/teamscale/report/testwise/jacoco/analysis/CachingClassAnalyzer.java

```diff
@@ -22,20 +26,26 @@
 	 * Creates a new analyzer that builds coverage data for a class.
 	 *
 	 * @param classCoverageLookup cache for the class' probes
+	 * @param coverage
+	 *            coverage node for the analyzed class data
+	 * @param stringPool
+	 *            shared pool to minimize the number of {@link String} instances
 	 */
-	public CachingClassAnalyzer(ClassCoverageLookup classCoverageLookup) {
+	public CachingClassAnalyzer(ClassCoverageLookup classCoverageLookup, ClassCoverageImpl coverage, StringPool stringPool) {
+		super(coverage, null, stringPool);
 		this.classCoverageLookup = classCoverageLookup;
 	}
 
 	@Override
 	public void visitSource(String source, String debug) {
+		super.visitSource(source, debug);
 		classCoverageLookup.setSourceFileName(source);
 	}
 
 	@Override
 	public MethodProbesVisitor visitMethod(final int access, final String name,
 										   final String desc, final String signature, final String[] exceptions) {
-		return new CachingMethodAnalyzer(classCoverageLookup);
+		return new CachingMethodAnalyzer(classCoverageLookup, Filters.ALL, this);
```

Comment:
Base class overrides visitEnd of MethodAnalyzer and modifies the base class coverage field. Do we no longer need this behavior?

Reply:
No we never had it. It appends the coverage produced by the analyzed method to the ClassCoverageImpl object. We store all the info we need in ClassCoverageLookup instead.